### PR TITLE
Refactor tests: do assertions at http level

### PR DESCRIPTION
**Pull request by drscre**

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -3,72 +3,63 @@ package zooz
...
+		require.NoError(c.t, err)
+		defer func() { _ = r.Body.Close() }()
 	}
-
-	if string(reqBody) != string(expectedReqBody) {
-		c.t.Errorf("Invalid request body: %s", string(reqBody))
+	if c.expectedBodyJSON != "" {
+		require.JSONEq(c.t, c.expectedBodyJSON, string(body))
+	} else {
+		require.Equal(c.t, "", string(body))
```
Comment (maratori) [inline client_test.go]: nit:
```suggestion
		require.Empty(c.t, string(body))
```

File: client_test.go [hunk 2]
```diff
@@ -3,72 +3,63 @@ package zooz
...
+	if len(c.expectedURL) > 0 && c.expectedURL[0] == '/' {
+		c.expectedURL = ApiURL + c.expectedURL
 	}
+	require.Equal(c.t, c.expectedURL, r.URL.String())
 
-	reqBody, err := json.Marshal(reqObj)
-	if err != nil {
-		c.t.Fatalf("Marshal error: %s", err)
+	for k, v := range c.expectedHeaders {
+		require.Equal(c.t, v, r.Header.Get(k))
```
Comment (maratori) [inline client_test.go:35]: Now you assert that expected headers is a subset of actual headers. Don't you want to assert that they are equal?

**Review by maratori — APPROVED** [review]

File: client_test.go (hunk 2 above)
Comment (drscre) [reply client_test.go:35]: Generally, we are interested in endpoint-specific special headers, such as idempotency key, client user agent, client ip address.
Copy-pasting full list of headers for every test would be reduntant.
Adding them as defaults in httpClientMock is an option, but i decided that testing full list of headers once in `TestCall_WithApiResponse` is enough.

File: client_test.go (hunk 1 above)
Comment (drscre) [reply client_test.go]: fixed

File: client_test.go (hunk 2 above)
Comment (drscre) [reply client_test.go:35]: Added more headers for assertion in `TestCall_WithApiResponse`, including content type and api version,

**Pull request merged**